### ISSUE TITLE
Shorthand method definitions not autocompleting

### ISSUE THREAD

**Issue by Faleij** [nikhilkalige/docblockr#205]
Fails to autocomplete method documentation for `obj.test`:

``` javascript
const obj = {
    /**
     * 
     */
    test(arg) {
        return true;
    }
};
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Method_definitions

**Comment by mattgoucher** [comment]
+1

**Comment by ianwremmel** [comment]
Seeing the same issue. I'm pretty sure this used to work, so it seems like more of a bug than an enhancement.

**Comment by MoritzKn** [comment]
@ianwremmel you are right, I overlooked this at first.
See [211](https://github.com/nikhilkalige/docblockr/issues/211#issuecomment-242512718) for information about why this worked and don't works anymore.

**Comment by MoritzKn** [comment]
Honestly I just don't know how to do this.. I guess this is simply not possible without context and to get this context we need to parse the whole file.

An example:
This:
```js
// an object
var foo = {
  // a function declaration
  bar()
  // the functions body in the next line
  {
    // do stuff
  }
};
```
just  means something completely different than this:
```js
// a function call
bar() // <-- automatic semicolon insertion steps in
// a block statement
{
// do stuff
}
```

**Comment by ianwremmel** [comment]
What are you using for parsing?  I recently had to do some code parsing and dug into the babel toolset a bit. It's surprisingly easy to use.

**Comment by MoritzKn** [comment]
@ianwremmel It's all hand written code. It just detects the line with the function declaration and parses the line using a few crazy RegExes. It's probably not perfect, but I don't think we can easily change it.

**Comment by hypnoboutique** [comment]
+1 here for support of ES6 method syntax.

(Great plugin, btw... thank you!)

**Comment by ianwremmel** [comment]
Document This for vscode seems to handle things pretty nicely. I wonder if there's something there that can be ported to Atom.

**Comment by lukasoppermann** [comment]
Same issue here, if you have a solution and need somebody to test it, let me know.

**Comment by spectralsun** [comment]
+1 please support ES6 function declarations

**Comment by MoritzKn** [comment]
Final!
This was fixed with version `v0.9.2`.
It ignores the edge case where a function call is followed by a block statement described above, but I think this is not likely to be a big problem and it seams this issue is indeed a problem for many people.

**Comment by AlexKvazos** [comment]
Thank you very much, very awaited bug fix/feature!
